### PR TITLE
Additional TLC Finales

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
Content "Additional TLC Finales"

#### Purpose of change
Adds two new finale rooms to the TCL since some other stuff is happening soon 👀

Also Migo need some love.

#### Describe the solution

TODO:

- [x] Add the rooms
- [x] Make them spawn
- [x] Make the rewards

Adds two new unique items and support infrastructure for them.

**Iridescent Carapace Ballistic Plates**
These plates are living tissue teased into the form of armor inserts. They can ablate and shatter like normal ceramic plates however given time they also grow back (24h). **they can also become overgrown and become incredibly uncomfortable to wear**. They include a lab notes manual with the recipe to manually shave the plates down to a reasonable size.

Overgrown:
image

Standard:
image

Undergrown:
image

Completely wrecked:
image

and the manual:
image

and craft:
image

**Second Skin Fluid Under Armor**
A disgusting second skin of non-neutonian fluid. Found as a strange choker, when activated it covers most of your body in a shock absorbing fluid. **It is permanently filthy and needs to be repaired with blood and a proprietary feeding system**

Off:
image

On:
image

Repair Kit (repairing requires blood but it needs water as charges):
image

As well this PR adds two new materials (Unobtanium) its a not particularly special metal but can't be repaired. 
Non newtonian alien fluid which is the good stuff and is repaired with blood.

A new mob which is a captured mostly dead MIGO.

And adds a spawn for the MIGO Biotech.

#### Describe alternatives you've considered
Not adding cool new sci-fi items

#### Testing
Everything works and is tested.
Can't fast forward time to test plates, need to actually step through the turns. 

#### Additional context
Room 1 (plates):
image

Room 2 (suit):
image

Biotech Room (small modification of the east vaults):
image

### CODE REVIEW

**Comment by TealcOneill** [comment]
I'm curious, how does the second skin work with players that have the squeamish trait?

File: data/json/items/book/fabrication.json [hunk 1]
```diff
@@ -573,5 +573,23 @@
     "max_level": 4,
     "intelligence": 9,
     "time": "30 m"
+  },
+  {
+    "id": "recipe_migo_plate",
+    "type": "BOOK",
+    "category": "manuals",
+    "name": { "str": "WX010 Living Armor Observations", "str_pl": "copies of WX010 Living Armor Observations" },
+    "description": "This sheaf of papers seems to describe maintenance on a new form of ballistic armor.  \"the regenerative plates though promising for field use do require maintenance or combat to stop from becoming overgrown\"",
```
Comment (casswedson) [inline data/json/items/book/fabrication.json:582]:
```suggestion
    "description": "This sheaf of papers seems to describe maintenance on a new form of ballistic armor.  \"the regenerative plates though promising for field use do require maintenance or combat to stop from becoming overgrown\".",
```

**Comment by bombasticSlacks** [comment]
> I'm curious, how does the second skin work with players that have the squeamish trait?

I thought it would work, that your character needs to remove them however that seems to not be the case. I'll fix it in a separate PR.

**Comment by Jarewill** [comment]
Oooo, this looks interesting.
However I have two questions regarding the second skin item.

For one, the odd choker is on the strapper layer, while the actual second skin is on the skin layer.
Seeing as the items transform into each other, it might cause annoyance having to move the item from the outermost layer to the innermost layer. (Unless the layer moving is automatic)
Though it still might seem odd that the strapped layer choker just disappears when the skin activates.

Another question I have regarding it is if it can be taken off once active.
I don't know how exactly this item functions, but if it's really a liquid, would it really be able to be taken off?

The ability to repair it when it's inactive might also be useful.

**Comment by bombasticSlacks** [comment]
>For one, the odd choker is on the strapper layer, while the actual second skin is on the skin layer.
Seeing as the items transform into each other, it might cause annoyance having to move the item from the outermost layer to the innermost layer. (Unless the layer moving is automatic)
Though it still might seem odd that the strapped layer choker just disappears when the skin activates.

I'm going to rearrange this so the item is a gorget and will cover the shoulders and be skintight even when off to avoid the rearranging.

>Another question I have regarding it is if it can be taken off once active.
I don't know how exactly this item functions, but if it's really a liquid, would it really be able to be taken off?

I debated making custom code for it transforming back when taken off but it seems like it would just confuse players. it doesn't behave exactly like liquid so if you wanted to pull the suit off while it was deployed I imagine the non-neutonian fluid would just congeal and be gross but still behave similar to a soft fabric. 

>The ability to repair it when it's inactive might also be useful.

It means you could repair it with copper which I don't want, however I think I have a fix for that planned.

**Comment by bombasticSlacks** [comment]
> I can fully understand that, I wouldn't want such a unique item to be repairable with just copper either.
I know there's the "repairs_like" property for items, but I don't know how exactly it works.

thats for items with no crafting recipe so that you can give them an associated repair difficulty. So it's more saying "this is about as hard to repair as X". What I'm going to do is move the material from copper to an alien metal with no repair options and that will fix it.

File: data/json/items/armor/exotic.json [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        "specifically_covers": [ "torso_lower", "arm_upper_l", "arm_elbow_l", "arm_lower_l", "arm_upper_r", "arm_elbow_r", "arm_lower_r" ],
+        "coverage": 100,
+        "encumbrance": 2
+      },
+      { "covers": [ "leg_l", "leg_r", "foot_l", "foot_r", "hand_l", "hand_r" ], "coverage": 100, "encumbrance": 2 }
+    ],
+    "use_action": [
+      {
+        "target": "migo_skin_off",
+        "msg": "the necklace hums and the liquid gets sucked back into it.",
```
Comment (casswedson) [inline data/json/items/armor/exotic.json:169]:
```suggestion
        "msg": "The gorget hums and the liquid gets sucked back into it.",
```

File: data/json/items/armor/exotic.json [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    "price": 2500,
+    "price_postapoc": 400,
+    "material": [ "unobtanium" ],
+    "symbol": "[",
+    "looks_like": "silver_necklace",
+    "color": "white",
+    "use_action": {
+      "need_worn": true,
+      "target": "migo_skin",
+      "msg": "murky ooze begins to leak out of the necklace and engulf your body from the neck down!",
```
Comment (casswedson) [inline data/json/items/armor/exotic.json:195]:
```suggestion
      "msg": "Murky ooze begins to leak out of the gorget and engulf your body from the neck down!",
```

File: data/json/items/armor/exotic.json [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+    "volume": "1533 ml",
+    "price": 0,
+    "price_postapoc": 0,
+    "material": [ "flesh" ],
+    "symbol": ",",
+    "color": "dark_gray",
+    "material_thickness": 4,
+    "//": "transforms over a day",
+    "countdown_interval": 86400,
+    "countdown_action": { "type": "transform", "target": "migo_plate_undergrown", "msg": "The iridescent carapace plate expands as it grows." },
```
Comment (casswedson) [inline data/json/items/armor/exotic.json:111]:
```suggestion
    "countdown_action": { "type": "transform", "target": "migo_plate_undergrown", "msg": "The plate of scarred flesh expands as it grows." },
```

**Comment by Raikiri** [comment]
I encountered everything added in this PR and it's awesome. I wish CDDA had much much more weird and rare things like this.

**Pull request merged**